python/lsst/sims/maf/driver/runBundle.py

```diff
@@ -0,0 +1,96 @@
+import os, re, copy, warnings
+import lsst.sims.maf.sliceMetrics as sliceMetrics
+import lsst.sims.maf.db as db
+import lsst.sims.maf.utils as utils
+
+def runBundle(mafBundle, verbose=True, makePlots=True):
+    """
+    mafBundle should be a dict with the following key:value pairs:
+    'metricList':List of MAF metric objects
+    'slicer': instatiated MAF slicer
+    'dbAddress': string that gives the address to the desired DB
+    'sqlWhere': string that gives the sql where clause for pulling the data
+    optional keys:
+    'outDir': Output directory
+    'stackerList': List of configured MAF stackers
+    'metadata': string containing metadata
+    'runName': string containing runName
+
+    plotOnly:  Restore the metric values anr re-plot with new plotkwargs XXX-todo
+    """
+
+    # Check required input set
+    reqKeys= ['metricList','slicer','dbAddress', 'sqlWhere']
+    for key in reqKeys:
+        if key not in mafBundle.keys():
+            raise ValueError('"%s" not found and required as a key on the input dictionary.' % key)
+
+    # Set the optional keys with defaults if missing
+    if 'outDir' not in mafBundle.keys():
+        mafBundle['outDir'] = 'Output'
+    if 'metadata' not in mafBundle.keys():
+        mafBundle['metadata'] = ''
+    if 'stackerList' not in mafBundle.keys():
+        mafBundle['stackerList'] = None
+    if 'runName' not in mafBundle.keys():
+        runName = re.sub('.*//', '', mafBundle['dbAddress'])
+        runName = re.sub('\..*', '', runName)
+        mafBundle['runName'] = runName
+
+    sm = sliceMetrics.RunSliceMetric(outDir = mafBundle['outDir'])
+
+    # Need to deepcopy here so we don't unexpectedly persist changes in mafBundle.
+    # Otherwise, plotDict can get set and persist.
```

Comment:
I'm not sure a deep copy is the best way to do this.
Why would a metric be re-used and NOT get its plotDict set appropriately in each use? 
Should we be enforcing/encouraging non-reuse of metric objects some other manner?